spec/puppet-server-lib/puppet/jvm/config_spec.rb

```diff
@@ -4,16 +4,53 @@
 
 describe Puppet::Server::Config do
   context "SSL context" do
-    it "is configured with :localcacert and not :cacert" do
+    before :each do
       Puppet[:hostcert] = "spec/fixtures/localhost-cert.pem"
       Puppet[:hostprivkey] = "spec/fixtures/localhost-privkey.pem"
-      Puppet[:cacert] = "SHOULD/NOT/BE/USED"
       Puppet[:localcacert] = "spec/fixtures/ca-cert.pem"
+    end
+
+    before :each do
+      Puppet::Server::Config.instance_variable_set(:@puppet_and_system_ssl_context, nil)
+      Puppet::Server::Config.instance_variable_set(:@puppet_only_ssl_context, nil)
+    end
+
+    it "is configured with :localcacert and not :cacert" do
+      Puppet[:cacert] = "SHOULD/NOT/BE/USED"
+
       begin
         Puppet::Server::Config.ssl_context
       rescue Java::JavaIO::FileNotFoundException
         fail 'SSL context configured with :cacert'
       end
     end
+
+    it "ssl_context and puppet_only_ssl_context are the same SSLContext" do
+      ssl_context = Puppet::Server::Config.ssl_context
+      puppet_only_ssl_context = Puppet::Server::Config.puppet_only_ssl_context
+
+      expect(ssl_context).to be_a_kind_of(Java::JavaxNetSsl::SSLContext)
+      expect(ssl_context).to eq(puppet_only_ssl_context)
+    end
+
+    it "loads the puppet certs and any existing configured system certs w/o error" do
+      Puppet[:ssl_trust_store] = "spec/fixtures/puppet-cacerts"
+      begin
+        ssl_context = Puppet::Server::Config.puppet_and_system_ssl_context
+        expect(ssl_context).to be_a_kind_of(Java::JavaxNetSsl::SSLContext)
+      rescue => e
+        fail "Could not initialize puppet_and_system_ssl_context because of #{e.inspect}"
+      end
+    end
+
+    it "warns if :ssl_trust_store is set but not readable" do
+      Puppet[:ssl_trust_store] = "spec/fixtures/foo.pem"
+      if ! File.exist?(Puppet::Server::Config::PUPPET_KEYSTORE_LOCATION)
+        expect(Puppet).to receive(:warning).with(/Could not find Puppet vendored keystore/)
+      end
```

Comment:
No, this is because we check in the code for the existence of a file path the agent should deliver. Whether or not this is received depends on whether or not a Puppet Agent is installed on the computer running the tests.

Reply:
I hadn't gone that route because I thought Ruby/RSpec had trouble resetting mocks on the File class and that Puppet proper had done a big effort years ago to remove them. But that, like the "symbols as truthy values" is probably an anachronism. Even if they aren't the best this test suite is so small that it probably won't hurt anything.